swig/include/python/gdal_python.i

```diff
@@ -863,6 +1015,73 @@ CPLErr ReadRaster1( double xoff, double yoff, double xsize, double ysize,
                    resample_alg=gdalconst.GRIORA_NearestNeighbour,
                    callback=None,
                    callback_data=None):
+        """
+        Write the contents of a NumPy array to a Dataset.
+
+        Parameters
+        ----------
+        array : np.ndarray
+            Two- or three-dimensional array containing values to write
+        xoff : int, default=0
+           The pixel offset to left side of the region of the band to
+           be written. This would be zero to start from the left side.
+        yoff : int, default=0
+           The line offset to top side of the region of the band to
+           be written. This would be zero to start from the top side.
+        band_list : list, optional
+            Indexes of bands to which data should be written. By default,
+            it is assumed that the Dataset contains the same number of
+            bands as levels in ``array``.
+        interleave : str, default="band"
+            Interleaving, "band" or "pixel". For band-interleaved writing,
+            ``array`` should have shape ``(nband, ny, nx)``. For pixel-
+            interleaved-writing, ``array`` should have shape
+            ``(ny, nx, nbands)``.
+        resample_alg : int, default = :py:const:`gdal.GRIORA_NearestNeighbour`
```

Comment:
it could potentially be used if xoff or yoff are not integers, but that's admitedly uncommon

Reply:
Can you point me to an example where this would have an effect? Writing an array at higher resolution than the band gives an error:

```
ds = gdal.GetDriverByName("MEM").Create("", 3, 3, 1, gdal.GDT_Float32)
ds.WriteArray(np.arange(25).reshape(5, 5), xoff=0.5, yoff=0.5)
# Traceback (most recent call last):
#   File "<stdin>", line 1, in <module>
#   File "/home/dan/dev/build-gdal-Desktop-Debug/swig/python/osgeo/gdal.py", line 4805, in WriteArray
#     data will be read from all bands.
#   File "/home/dan/dev/build-gdal-Desktop-Debug/swig/python/osgeo/gdal_array.py", line 542, in DatasetWriteArray
#     return BandWriteArray(ds.GetRasterBand(band_list[0]),
#  File "/home/dan/dev/build-gdal-Desktop-Debug/swig/python/osgeo/gdal_array.py", line 666, in BandWriteArray
#     raise ValueError("array larger than output file, or offset off edge")
# ValueError: array larger than output file, or offset off edge
```

and when writing an array of lower resolution, I don't see an effect of this parameter
```
ds.WriteArray(np.array([[1]]), xoff=0.5, yoff=0.5, resample_alg=gdal.GRIORA_Average)
# 0
ds.ReadAsArray()
# array([[0., 0., 0.],
#        [0., 1., 0.],
#        [0., 0., 0.]], dtype=float32)
```

(I would have expected 
```
array([[0.25, 0.25, 0.  ],
       [0.25, 0.25, 0.  ],
       [0.  , 0.  , 0.  ]])
```

Reply:
> Can you point me to an example where this would have an effect?

not really. Actually looking at the generic logic in GDALRasterBand;;IRasterIO() in gcore/rasterio.cpp at lines 610-732 the resampling method is ignored. So it is more about a theoretical capability than something actually implemented

Reply:
Updated the docstring to note that this is a placeholder argument.